hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3ABlockOutputArray.java

```diff
@@ -107,13 +108,18 @@ public void testDiskBlockCreate() throws IOException {
         "very_long_s3_key__very_long_s3_key__very_long_s3_key__very_long_s3_key__" +
         "very_long_s3_key__very_long_s3_key__very_long_s3_key__very_long_s3_key__" +
         "very_long_s3_key";
-    S3ADataBlocks.DataBlock dataBlock = diskBlockFactory.create("spanId", s3Key, 1,
-      getFileSystem().getDefaultBlockSize(), null);
-    LOG.info(dataBlock.toString()); // block file name and location can be viewed in failsafe-report
-
-    // delete the block file
-    dataBlock.innerClose();
-    diskBlockFactory.close();
+    long blockSize = getFileSystem().getDefaultBlockSize();
+    try (S3ADataBlocks.BlockFactory diskBlockFactory =
+           new S3ADataBlocks.DiskBlockFactory(getFileSystem());
+         S3ADataBlocks.DataBlock dataBlock =
+           diskBlockFactory.create("spanId", s3Key, 1, blockSize, null);
+    ) {
+      boolean created = Arrays.stream(
+        Objects.requireNonNull(new File(getConfiguration().get("hadoop.tmp.dir")).listFiles()))
+          .anyMatch(f -> f.getName().contains("very_long_s3_key"));
+      assertTrue(created);
```

Comment:
add a message to print if the assert is false: we need to be able to start debugging without having to work back from the first line of the stack trace as to what went wrong. include that hadoop.tmp.dir value in the message too